### PR TITLE
Pb 36

### PR DESCRIPTION
**Pull request by CharlieTono**
Add new CountryCode entity

### CODE REVIEW

File: phonebook-api/src/main/java/com/telran/phonebookapi/model/CountryCode.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@NoArgsConstructor
+@Getter
+public class CountryCode {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (sharf85) [inline phonebook-api/src/main/java/com/telran/phonebookapi/model/CountryCode.java:17]: make the code as id, remove @GeneratedValue

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/CountryCodeController.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import com.telran.phonebookapi.mapper.CountryCodeMapper;
+import com.telran.phonebookapi.service.CountryCodeService;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@CrossOrigin
```
[the remark this answers is not in the document]
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/controller/CountryCodeController.java]: removed

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/CountryCodeController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+@RequestMapping("/api/code")
```
[the remark this answers is not in the document]
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/controller/CountryCodeController.java]: /api/country_code

File: phonebook-api/src/main/java/com/telran/phonebookapi/controller/CountryCodeController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    public CountryCodeDto getById(@PathVariable int id) {
+        return countryCodeMapper.mapCountryCodeToDto(countryCodeService.getById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    public void removeById(@PathVariable int id) {
+        countryCodeService.removeById(id);
+    }
+
+    @GetMapping("/all")
```
[the remark this answers is not in the document]
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/controller/CountryCodeController.java]: removed

File: phonebook-api/src/main/java/com/telran/phonebookapi/mapper/CountryCodeMapper.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+@Component
+public class CountryCodeMapper {
+
+    public CountryCodeDto mapCountryCodeToDto(CountryCode countryCode) {
+        return new CountryCodeDto(countryCode.getId(), countryCode.getCode(), countryCode.getCountry());
+    }
+
+    public CountryCode mapDtoToCountryCode(CountryCodeDto countryCodeDto) {
+        return new CountryCode(countryCodeDto.id, countryCodeDto.code, countryCodeDto.country);
```
[the remark this answers is not in the document]
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/mapper/CountryCodeMapper.java]: removed

File: phonebook-api/src/main/java/com/telran/phonebookapi/mapper/CountryCodeMapper.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+public class CountryCodeMapper {
+
+    public CountryCodeDto mapCountryCodeToDto(CountryCode countryCode) {
+        return new CountryCodeDto(countryCode.getId(), countryCode.getCode(), countryCode.getCountry());
+    }
+
+    public CountryCode mapDtoToCountryCode(CountryCodeDto countryCodeDto) {
```
[the remark this answers is not in the document]
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/mapper/CountryCodeMapper.java]: removed

File: phonebook-api/src/main/java/com/telran/phonebookapi/service/CountryCodeService.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    static final String CODE_EXIST = "Error! This country code has already existed in our DB";
+    static final String CODE_DOES_NOT_EXIST = "Error! This country code doesn't exist in our DB";
+
+    ICountryCodeRepository countryCodeRepository;
+
+    public CountryCodeService(ICountryCodeRepository countryCodeRepository) {
+        this.countryCodeRepository = countryCodeRepository;
+    }
+
+    public void add(int id, String code, String country) {
```
[the remark this answers is not in the document]
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/service/CountryCodeService.java]: removed

File: phonebook-api/src/main/java/com/telran/phonebookapi/model/CountryCode.java (hunk 1 above)
Comment (CharlieTono) [reply phonebook-api/src/main/java/com/telran/phonebookapi/model/CountryCode.java]: GeneratedValue was removed, int code is using as "Id".

File: phonebook-api/src/main/java/com/telran/phonebookapi/model/Phone.java [hunk 8]
```diff
@@ -17,9 +17,9 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private int id;
     @Setter
-    private String countryCode;
+    private int countyCode;
```
Comment (sharf85) [inline phonebook-api/src/main/java/com/telran/phonebookapi/model/Phone.java:20]: spell check

**Review by sharf85 — CHANGES_REQUESTED** [review]

**Pull request merged**